spec-dtslint/operators/count-spec.ts

```diff
@@ -0,0 +1,14 @@
+import { of } from 'rxjs';
+import { count } from 'rxjs/operators';
+
+it('should always infer number', () => {
+  const o = of(1, 2, 3).pipe(count(x => x > 1)); // $ExpectType Observable<number>
```

Comment:
I believe we should also test the predicate overloads.

```ts
predicate: (value: T, index: number, source: Observable<T>)
```

Reply:
Added few more tests around the predicate overloads types. 
Please let me know if they are of a good value. 
Thanks